back/engines/commercial/multi_tenancy/lib/tasks/core/setup_and_support.rake

```diff
@@ -245,13 +245,23 @@ namespace :setup_and_support do
     end
   end
 
-  desc 'Delete users and reactions'
-  task :delete_users_reactions, %i[host url] => [:environment] do |_t, args|
+  desc 'Delete spam users (from email list) and their contributions'
+  task :delete_users_participation, %i[host url] => [:environment] do |_t, args|
     emails = open(args[:url]).readlines.map(&:strip)
     Apartment::Tenant.switch(args[:host].tr('.', '_')) do
       users = User.where email: emails
+      Initiative.where(author: users).destroy_all
+      Idea.where(author: users).destroy_all
       Reaction.where(user: users).destroy_all
+      Comment.where(author: users).destroy_all
+      Basket.where(user: users).destroy_all
       users.destroy_all
```

Comment:
Just want to make sure - calling the `users.destroy_all` before the `users.each do` is fine right?

Also just a question - what's the difference between the `users.destroy_all` and the `users.each do` block of code?

Reply:
Good catch, I forgot to remove `users.destroy_all`. I'm replacing it with sidefx, so we can remove them from external services (we should probably respect the privacy of spam users too).